### PR TITLE
Add displaying cycle dots before agent reply in AS studio

### PR DESCRIPTION
**Pull request by qbc2016**
## Description

Add a decorator to the reply function. In AS studio, each agent will loop through “....” before replying, indicating that it is thinking.

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has passed all tests
- [ ]  Docstrings have been added/updated in Google Style
- [ ]  Documentation has been updated
- [ ]  Code is ready for review

### CODE REVIEW

**Comment by pan-x-c** [comment]
Using decorators on the `reply` method will reduce execution efficiency.

File: src/agentscope/agents/agent.py [hunk 1]
```diff
@@ -13,6 +15,51 @@
...
+            *args (Any): Variable length argument list for the decorated
+            method.
+            **kwargs (Any): Arbitrary keyword arguments for the decorated
+            method.
+
+        Returns:
+            Any: The return value from the decorated method.
+        """
+        if hasattr(thread_local_data, "uid"):
+            time.sleep(0.1)
```
Comment (pan-x-c) [inline src/agentscope/agents/agent.py:55]: remove sleep.
Using sleep here will cause a time delay in model calling, further increasing the running time.

File: src/agentscope/agents/agent.py [hunk 2]
```diff
@@ -23,6 +70,12 @@ def __call__(cls, *args: tuple, **kwargs: dict) -> Any:
         instance._init_settings = {"args": args, "kwargs": kwargs}
         return instance
 
+    def __new__(mcs, name: Any, bases: Any, attrs: Any) -> Any:
+        if "reply" in attrs and not attrs.get("_skip_reply_decorator", False):
+            original_reply = attrs["reply"]
+            attrs["reply"] = pre_reply_decorator(original_reply)
```
Comment (pan-x-c) [inline src/agentscope/agents/agent.py:76]: Seriously affects performance, it is best not to do this

**Review by ZiTao-Li — APPROVED** [review]
LGTM

**Pull request merged**